pdns/pdnsutil.cc

```diff
@@ -4407,96 +4407,255 @@ static int backendLookup(vector<string>& cmds)
+  "Catalog",
+  "Metadata",
+  "Zone",
+  "RRSet",
+  "DNSSEC",
+  "CDS/CDNSKEY",
+  "NSEC3",
+  "TSIG key",
+  "Zone key",
+  "Other"
+};
+
+struct commandDispatcher {
+  bool requiresInitialization; // need to invoke reportAllTypes() before handler
+  int (*handler)(std::vector<std::string>&);
+  commandGroup group;
+  const std::string_view help;
+};
+
+// clang-format off
+static const std::unordered_map<std::string, commandDispatcher> commands{
+  {"activate-tsig-key", {true, activateTSIGKey, GROUP_TSIGKEY,
+   "activate-tsig-key ZONE NAME {primary|secondary|producer|consumer}\n"
+   "                                   Enable TSIG authenticated AXFR using the key NAME for ZONE"}},
+  {"activate-zone-key", {true, activateZoneKey, GROUP_ZONEKEY,
+   "activate-zone-key ZONE KEY-ID      Activate the key with key id KEY-ID in ZONE"}},
+  {"add-autoprimary", {true, addAutoprimary, GROUP_AUTOPRIMARY,
+   "add-autoprimary IP NAMESERVER [account]\n"
+   "                                   Add a new autoprimary "}},
+  {"add-meta", {true, setMeta, GROUP_META,
+   "add-meta ZONE KIND VALUE           Add zone metadata, this adds to the existing KIND\n"
+   "                   [VALUE ...]"}},
+  {"add-record", {true, addRecord, GROUP_ZONE,
+   "add-record ZONE NAME TYPE [ttl] content\n"
+   "             [content..]           Add one or more records to ZONE"}},
+  {"add-zone-key", {true, addZoneKey, GROUP_ZONEKEY,
+   "add-zone-key ZONE {zsk|ksk} [BITS] [active|inactive] [published|unpublished]\n"
+   "             [rsasha1|rsasha1-nsec3-sha1|rsasha256|rsasha512|ecdsa256|ecdsa384"
+#if defined(HAVE_LIBSODIUM) || defined(HAVE_LIBCRYPTO_ED25519)
+   "|ed25519"
+#endif
+#if defined(HAVE_LIBCRYPTO_ED448)
+   "|ed448"
+#endif
+   "]\n"
+   "                                   Add a ZSK or KSK to zone and specify algo&bits"}},
+  {"b2b-migrate", {true, B2BMigrate, GROUP_OTHER,
+   "b2b-migrate OLD NEW                Move all data from one backend to another"}},
+  {"backend-cmd", {true, backendCmd, GROUP_OTHER,
+   "backend-cmd BACKEND CMD [CMD..]    Perform one or more backend commands"}},
+  {"backend-lookup", {true, backendLookup, GROUP_OTHER,
+   "backend-lookup BACKEND NAME [[TYPE] CLIENT-IP-SUBNET]\n"
+   "                                   Perform a backend lookup of NAME, TYPE and CLIENT-IP-SUBNET"}},
+  {"bench-db", {true, benchDb, GROUP_OTHER,
+   "bench-db [filename]                Bench database backend with queries, one zone per line"}},
+  {"change-secondary-zone-primary", {true, changeSecondaryZonePrimary, GROUP_ZONE,
+   "change-secondary-zone-primary ZONE primary-ip [primary-ip..]\n"
+   "                                   Change secondary zone ZONE primary IP address to primary-ip"}},
+  {"check-all-zones", {true, checkAllZones, GROUP_ZONE,
+   "check-all-zones [exit-on-error]    Check all zones for correctness. Set exit-on-error to exit immediately\n"
+   "                                   after finding an error in a zone."}},
```

Comment:
```suggestion
   "check-all-zones [exit-on-error]    Check all zones for correctness. Use exit-on-error to exit immediately\n"
   "                                   upon finding the first error in any zone."}},
```

Is this what it does?